consensus/src/state_computer.rs

```diff
@@ -222,17 +222,38 @@ impl StateComputer for ExecutionProxy {
                 "Got state compute result, post processing."
             );
             let pipeline_execution_result = fut.await?;
-            let input_txns = pipeline_execution_result.input_txns.clone();
+            let user_txns = &pipeline_execution_result.input_txns;
             let result = &pipeline_execution_result.result;
 
             observe_block(timestamp, BlockStage::EXECUTED);
 
-            // notify mempool about failed transaction
-            if let Err(e) = txn_notifier.notify_failed_txn(input_txns, result).await {
-                error!(
-                    error = ?e, "Failed to notify mempool of rejected txns",
-                );
+            let compute_status = result.compute_status_for_input_txns();
+            // the length of compute_status is user_txns.len() + num_vtxns + 1 due to having blockmetadata
+            if user_txns.len() >= compute_status.len() {
+                // reconfiguration suffix blocks don't have any transactions
+                // otherwise, this is an error
+                if !compute_status.is_empty() {
+                    error!(
+                        "Expected compute_status length and actual compute_status length mismatch! user_txns len: {}, compute_status len: {}, has_reconfiguration: {}",
+                        user_txns.len(),
+                        compute_status.len(),
+                        result.has_reconfiguration(),
+                    );
+                }
+            } else {
+                let user_txn_status = &compute_status[compute_status.len() - user_txns.len()..];
```

Comment:
does state checkpoint have compute status?

Reply:
state checkpoint is not part of the compute_status_for_input_txns